### PR TITLE
feat: generate third party license notices and append to existing

### PR DESCRIPTION
**Pull request by ts-amz**
# Description
Created a simple script to generate third party license notices and append them to our existing notices.

## Type of change

- [ ] *Bug fix (non-breaking change which fixes an issue)*
- [ ] *New feature (non-breaking change which adds functionality)*
- [ ] *Breaking change (fix or feature that would cause existing functionality to not work as expected)*
- [ ] *Refactor* (existing code being refactored)
- [ ] *This change includes a documentation update*

# Submission Checklist

- [ ] CI dry-run passing
- [ ] Integration tests passing locally
- [ ] Change logs generated

### CODE REVIEW

File: deployment/generate-attributions.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# generate or update the THIRD-PARTY-NOTICES
+# requires oss-attribution-generator - https://www.npmjs.com/package/oss-attribution-generator
+# appends the generated list to the base file that specifies rush attributions
+# must be run from the deployment/ directory where this resides
+
+cat ./third-party-notices-base.txt > ../THIRD-PARTY-NOTICES.txt
```
Comment (canavandl) [inline deployment/generate-attributions.sh:6]: i'm confused how this works. Does THIRD-PARTY-NOTICES just grow monotonically forever?

Comment (ts-amz) [reply deployment/generate-attributions.sh:6]: No, this first step overwrites it with the base, then the subsequent step appends to it. It's a new file every time. But I couldn't figure out how this base was derived, so I'm not clearing it and instead only adding to it (the base is what was in THIRD-PARTY-NOTICES.txt before this change).

Comment (canavandl) [reply deployment/generate-attributions.sh:6]: ah you're right. I confused single ">" with ">>" (to append)

**Review by canavandl — APPROVED** [review]
LGTM

**Review by pcozzi — APPROVED** [review]

**Pull request merged**